lib/cn/vblock_builder.c

```diff
@@ -310,10 +310,25 @@ vbb_finish(struct vblock_builder *bld, struct blk_list *vblks)
     return 0;
 }
 
-void
+merr_t
 vbb_set_agegroup(struct vblock_builder *bld, enum hse_mclass_policy_age age)
 {
+    merr_t                    err;
+    struct mclass_policy *    policy;
+    struct mpool_mclass_props props;
+
     bld->agegroup = age;
+
+    policy = cn_get_mclass_policy(bld->cn);
+
+    err = mpool_mclass_props_get(
+        bld->ds, policy->mc_table[bld->agegroup][HSE_MPOLICY_DTYPE_KEY], &props);
+    if (err)
+        return err;
+
+    bld->max_size = props.mc_mblocksz;
+
+    return err;
```

Comment:
I think I see a good many calls in spill.c that do not check this new return code (via kvset_builder_set_agegroup).  Maybe I missed something?
What does it mean if you set the agegroup but the props-get call fails?  The agegroup has been modified, but the max_size has not..

Reply:
Also, that should not be DTYPE_KEY.

Reply:
Maybe @nabeelmmd can fix the DTYPE in his PR

Reply:
Yes, will fix it in my PR.